### DISCUSSION TITLE
建议增加开屏广告通用跳过规则

### DISCUSSION THREAD

**Discussion by jiuqianyuan** [gkd-kit/gkd#217] [Ideas]
- #48 
- #79 

### 1、建议增加类似李跳跳的 通用开屏规则 适配所有app
比如在所有app的开屏界面自动匹配以下规则，可跳过大部分app的开屏广告。
` [text *="跳过" || text *="关闭" || desc *="跳过" || desc *="关闭"]`

另在[gkd-kit/subscription](https://github.com/gkd-kit/subscription)库中检索到[52款应用](https://github.com/search?q=repo%3Agkd-kit%2Fsubscription%20tt_splash_skip_btn&type=code)的开屏广告貌似使用了同一个第三方框架 ，所以是否将以下规则也列为 通用开屏规则
`[id $= "id/tt_splash_skip_btn"]`

### 2、是否考虑开发类似李跳跳的自定义添加规则图形界面

不过这样相比目前的统一编写规则模式，好像有些混乱

### **随便讨论**

![222](https://github.com/gkd-kit/gkd/assets/39406781/0b7b84d5-2c34-4b8e-978c-c758aed41f8e)

**Comment by boplay** [comment]
我也觉得很有必要，用了很久的李跳跳，发现GKD能实现的功能比李跳跳更多。以前用李跳跳可以用Auto.js自己加规则，大多数APP的广告和弹窗都能解决。没学过软件，尝试学习编写GKD规则，范例里很多命令不知道是什么意思，无从下手。
比如https://github.com/gkd-kit/subscription/blob/main/CONTRIBUTING.md教程里面
        activityIds: ['cn.wps.moffice.main.StartPublicActivity', 'cn.wps.moffice.main.local.HomeRootActivity'],
其中activityIds  、.main.StartPublicActivity、.HomeRootActivity完全不知道是什么意思。
还有https://github.com/gkd-kit/subscription/blob/main/src/apps/io.sbyd.app.ts里的
        'TextView < LinearLayout[childCount=2] < FrameLayout[childCount=2] > FrameLayout[childCount=1] > ImageView',也不知道是什么意思。
对外行来说，可能要花很多功夫学习才能编写出规则，也建议增加通用规则。

**Comment by QinRay** [comment]
可以参考一下tasker系列软件autotools 这个工具 他的工作原理是监测用户点击页面位置 获取相关元素的信息 这样就能实现自动点击 

gkd我用的感觉是功能强大 但是不能自己定制 或者说对于大多数人来说 难度较大 可以参考一下autotools 让用户进行模拟点击 获取相关信息 然后用户就能定制了

![Screenshot_20231128-150116.png](https://github.com/gkd-kit/gkd/assets/74906051/af6803e4-c946-45c9-9697-e919ce244f7e)

**Comment by no-swing** [comment]
我用繁体中文系统，会有极个别app的跳过是繁体中文，‘text *="跳过 ’ 这种匹配方式会有漏掉的

**Comment by lisonge** [comment]
https://github.com/gkd-kit/gkd/releases
